### ISSUE TITLE
Multiple server IP addresses to connect to as a list

### ISSUE THREAD

**Issue by nbolton** [deskflow/deskflow#3665]
**Imported issue:**
- Author: Ehryk Menze
- Date: 2013-09-19 05:33:47
- Legacy ID: 3735

**Detail:**

I'd like to be able to enter multiple server addresses for my clients to connect to. I use my laptop as a client at work and at home - to connect to two different servers with completely different IP addresses and names.

**Example:** _Server IP: 192.168.1.12,10.104.12.92,PC469_

Expected: Allow a , or ; separated list of IP addresses to search for Synergy servers. If more than one are available, it should take the first listed as having higher priority.

Optional: Allow the selection of 'continue polling while connected' (switching to a higher-priority server if it appears while connected to a lower-priority one) optionally, and if disabled only check for servers on disconnect.

**Versions and operating systems:**

All

**Temporary workarounds:**

Manually changing the client's IP address every time (twice/day for me). Annoying.

**Similar bugs:**

[Issue #388](http://synergy-foss.org/spit/issues/details/388/) is the desire for a command-line version of this.
#7889 - Similar based on location
#9058 

### Voting Instructions

React with :+1: or :-1: to vote on this request

**Comment by AaronLS** [comment]
@nbolton 

Hey, why was this closed?  Have the exact same scenario, going back from work to home with a client computer.  Would be nice to have "Server IP:" support a semicolon delimited list of computer names/servers. (not colon because that might confuse people who try to put a port number in there)

Not trying to connect multiple servers at same time, only one at any given time, but pain to have to modify config on client computer as it requires me connect mouse/keyboard when otherwise I wouldn't need to. (It's a all-in-one desktop that's portable, so no builtin mouse/keyboard)

**Comment by ajcollett** [comment]
Yes! This is exactly what I need too. I'd even be willing to add to my donation to have this feature ;-P

**Comment by sacmt-hciie** [comment]
Can one bind 2 synergy client instances to 2 different IP's... Docker?

**Comment by bumbummen99** [comment]
I also request this feature, using the paid version and it's just irritating to change the ip every time i want to use synergy at work or at home. this is a must have feature.

**Comment by ajcollett** [comment]
Work-around: Set the hostname on your DHCP's to be the same thing in both locations, then use hostname instead of IP. I do this, and it works happily. But, it is not a fix.

**Comment by bumbummen99** [comment]
i doubt that my boss will let me play around with our dhcp just for my little gadget so this will not work out for me. I require this feature very badly.

**Comment by dmdsig** [comment]
I require this too and surprised it doesnt have it out of the box, i work with 3 different computers and typing adress manually is a pain and sometimes the adresses change, would be useful if you could at least save ip and click and button.

**Comment by y4my4my4m** [comment]
Will this ever be implemented?
I don't mind building it but could anyone direct me roughly where in the code that would be?

**Comment by candeira** [comment]
This can be done by configuring avahi in both synergy servers so they have the same mDNS name.

**Comment by theclaw37** [comment]
So after 4 years, this is still not a feature ? Nice.

**Comment by y4my4my4m** [comment]
@theclaw37 ive been using a fork of Synergy called Barrier, they have SSL and this feature.

**Comment by theclaw37** [comment]
@y4my4my4m I cloned the repo, already started work to support this. Wish I knew this. Thanks.

**Comment by nbolton** [comment]
@y4my4my4m Are you sure IP as a list support was added? There doesn't appear to be an issue for it. https://github.com/debauchee/barrier/issues

I only had a quick glance at the commits, but there doesn't appear to be anything obvious... perhaps you could let me know the hash of the commit?

**Comment by plato1123** [comment]
Is this going to happen eventually?  Seems like from a technical standpoint it would be a really easy improvement.  At least in my case only one of the two server computers is on/available on the network at a time so if you have two IP addresses, connect to whichever one you can and you're good!

**Comment by GreenMan36** [comment]
This is a great feature to have and essential for some.

**Comment by JustinFarris** [comment]
Another use case for this: Sometimes-on VPN client won't allow local LAN connectivity, but local router won't hairpin LAN clients to inbound NAT forward rule. So I have to stop and start the synergy client with a different host/port whenever I toggle between VPN.

**Comment by kysoneast** [comment]
+1 for this, I regularly switch between LAN / WiFi but still need to control...

**Comment by pyralix** [comment]
I have a computer that's connected to another on a local /30 connection for the sake of latency, and another on the wifi. I achieved multiple listening IPs by using 0.0.0.0, which causes the agent to bind to all IP addresses.

**Comment by nbolton** [comment]
This sounds like a useful issue, and I'd love to see it implemented.

**Comment by jief666** [comment]
+1 for this.

**Comment by riyad** [comment]
Do I remember correctly that  if you use a DNS name Deskflow already tries all the returned IP addresses in order to establish a connection? I.e. there is logic to try multiple addresses until one succeeds ... it seems it's just not plugged through the full stack.

**Comment by sithlord48** [comment]
> Do I remember correctly that if you use a DNS name Deskflow already tries all the returned IP addresses in order to establish a connection? I.e. there is logic to try multiple addresses until one succeeds ... it seems it's just not plugged through the full stack.

We do not do any logic to try additional addresses, if you type in a name you should get the name resolved via DNS

edited for clarity

**Comment by jief666** [comment]
The Logic is : i regularly use 3 different servers with 3 different dns name and 3 different ip on 3 completely different networks with a laptop configured as client. It would be nice if I don’t have to change the name or ip of the server each time I change network.
Looks like I’m not the only one thinking that…

**Comment by sithlord48** [comment]
edit: updated the comment above

> The Logic is : i regularly use 3 different servers with 3 different dns name and 3 different ip on 3 completely different networks with a laptop configured as client. It would be nice if I don’t have to change the name or ip of the server each time I change network.
Looks like I’m not the only one thinking that…

This feature as described here will not happen what is likely to happen is some kind of "Profile" support. Where you will be able to selected a profile. While working on that we can discuss how roaming between profiles may work.

**Comment by jief666** [comment]
Problem with profile is that you need to click something on the client. Which means you need a physical mouse, just for selecting the profile.
The possibility to enter a small list of server (either name or IP address) seems simpler and doesn't bother the basic user (they won't even know it's there).
I have to admit that I don't understand what's the problem with that feature...

**Comment by sithlord48** [comment]
> Problem with profile is that you need to click something on the client. Which means you need a physical mouse, just for selecting the profile.

 Maybe profiles can have triggers based on the network ssid( if wifi) or ip if wired or network profile if using network manager or something else. changing of profiles does not have to be completely manual . That is what i ment by "While working on that we can discuss how roaming between profiles may work."

Also your client is moving around places i think its rather safe to assume its laptop, no ?

**Comment by jief666** [comment]
No it’s not a laptop.

What’s the problem about allowing few server  ip and/or name ?
